internal/osbuild2/qemu_stage.go

```diff
@@ -91,6 +100,26 @@ func (o VMDKOptions) validate() error {
 	if o.Type != QEMUFormatVMDK {
 		return fmt.Errorf("invalid format type %q for %q options", o.Type, QEMUFormatVMDK)
 	}
+
+	allowedVMDKSubformats := []VMDKSubformat{
+		VMDKSubformatMonolithicFlat,
+		VMDKSubformatMonolithicSparse,
+		VMDKSubformatTwoGbMaxExtentFlat,
+		VMDKSubformatTwoGbMaxExtentSparse,
+		VMDKSubformatStreamOptimized,
+		"", // default empty value when the option is not set
```

Comment:
Yeah, another option would be to make the variable a pointer to the type.

I used this, because I used it in other stage before https://github.com/osbuild/osbuild-composer/blob/a999b7b04c268de75cc4cd9a2f9c5f23c8c962a1/internal/osbuild2/dnf_automatic_config_stage.go#L44

Personally I find the pointer to the type cleaner than defining the default string variable value as a valid enum, since assigning the empty string value does not make much sense.

What do you think?

Reply:
Indeed, pointer is nice.  It fits the semantics of the value and the option much better.  `FormatNone` might send the wrong message because there's a default fallback format when it's not specified and `FormatDefault` is semantically correct, but doesn't provide the actual explicit default value.
Being able to actually omit the value makes it clear that we're not specifying one and letting the tool do whatever it does when it's not specified.

Reply:
So, using pointer turned out to be impractical. In Go, one can not easily get pointer to a basic type literal. We have helper functions in the `internal/common` package for some types, but because of using type alias, it is not possible to use the existing one for `string`. I would have to introduce a helper function specifically for this type, which seemed weird to me.

So I ended up using the approach from `tar` stage implementation which at least looks cleaner https://github.com/osbuild/osbuild-composer/blob/aaee939f9b3544e208a3e8fa79324b68c91c2d8a/internal/osbuild2/tar_stage.go#L47-L65

Reply:
Consistency \o/